### ISSUE TITLE
self._handle is not defined in __del__

### ISSUE THREAD

**Issue by bryanwweber** [drivendataorg/cloudpathlib#299]
Hello! Thank you for this awesome package. I've been debugging some errors in pytest fixtures lately, specifically `ClientMismatchError`s (which are my fault, not `cloudpathlib`). Due to causing this error, the `self._handle` attribute is never set, so when `pytest` tears the test down and calls `__del__` on `CloudPath`, there is a further exception raised which pytest leaves a note about. The fix is fairly simple, it should be enough to move the definition of the `self._handle` attribute to the top of `__init__`, or else use a `try/except` block in `__del__` to catch this situation. Are you open to a patch for this?

Sample traceback:

```
python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function CloudPath.__del__ at 0x1071b9940>

  Traceback (most recent call last):
    File ".../lib/python3.9/site-packages/cloudpathlib/cloudpath.py", line 184, in __del__
      if self._handle is not None:
  AttributeError: 'S3Path' object has no attribute '_handle'

    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
```

This is also documented in pytest here: https://docs.pytest.org/en/7.1.x/how-to/failures.html#warning-about-unraisable-exceptions-and-unhandled-thread-exceptions

**Comment by pjbull** [comment]
Thanks @bryanwweber! Yes, let's move the `_handle` initialization to near the top of that function so we don't hit this in the future. That would be a great patch.

**Comment by bryanwweber** [comment]
Thanks for the quick reply @pjbull! The fix is in #300.
